src/libs/actions/templates.ts

```diff
@@ -88,6 +89,26 @@ export async function fetchTemplateById(templateId: string) {
   }
 }
 
+export async function addGeneratedTemplateHistory({
+  templateId,
+  dataValue,
+}: GeneratedTemplates) {
+  if (!templateId) {
+    throw new Error('Missing templateId');
+  }
+
+  try {
+    await db.insert(generated_templates).values({
+      template_id: templateId,
+      data_value: dataValue || null,
+    });
+
+    return { message: 'History Added for Generated Template' };
+  } catch (error) {
+    throw new Error(`Failed to Create History: ${error}`);
```

Comment:
do not throw exception here.
Generation of PDF should not fail, just because we could not save it in our history, instead log with template id